corehq/apps/hqwebapp/static/hqwebapp/js/components/pagination.js

```diff
@@ -0,0 +1,37 @@
+hqDefine('hqwebapp/js/components/pagination', function() {
+    return {
+        viewModel: function(params){
+            var self = {};
+            self.currentPage = ko.observable(params.currentPage || 1);
+            self.totalItems = ko.observable(params.totalItems || 500);
+            self.perPage = ko.observable(params.perPage || 10);
+            self.numPages = ko.computed(function(){
+                return Math.ceil(self.totalItems() / self.perPage());
+            });
+
+            self.nextPage = function(){
+                self.currentPage(Math.min(self.currentPage() + 1, self.numPages()));
+            };
+            self.previousPage = function(){
+                self.currentPage(Math.max(self.currentPage() - 1, 1));
+            };
+            self.goToPage = function(page){
+                self.currentPage(page);
+            };
+            self.pagesShown = ko.computed(function(){
+                var pages = [];
+                for (var i = 1; i <= self.numPages(); i++){
```

Comment:
Nit: Would be nice to have fewer magic numbers in this loop, which would also make it easy to make the max number of shown pages a parameter (not 100% sure that's desirable, but the releases page could easily fit 10 instead of 5 without looking ridiculous). FWIW there's similar logic in the [dashboard tiles](https://github.com/dimagi/commcare-hq/commit/918fba3a2e5ae2540f7c4f98dd67ba43526824a2#diff-3a3eb4e19bc638248d5fc4a41efc16afR52).